dot/core/service.go

```diff
@@ -496,7 +496,10 @@ func (s *Service) ProcessBlockAnnounceMessage(msg network.Message) error {
 
 // ProcessBlockRequestMessage processes a block request message, returning a block response message
 func (s *Service) ProcessBlockRequestMessage(msg network.Message) error {
-	blockRequest := msg.(*network.BlockRequestMessage)
+	blockRequest, ok := msg.(*network.BlockRequestMessage)
```

Comment:
given the method is ProcessBlockRequestMessage, would be appropriate that the logic that validates the input is a helper placed outside the func, and the func signature is changed to accept only BlockRequestMessage. Would you agree ?

Reply:
yes, I agree, I'll update handleReceivedMessage to cast it beforehand